pkg/prometheus/operator.go

```diff
@@ -969,18 +968,22 @@ func (c *Operator) sync(key string) error {
 }
 
 func createSSetInputChecksum(p monitoringv1.Prometheus, c Config, ruleConfigMapNames []string) (string, error) {
-	json, err := json.Marshal(
-		struct {
-			P monitoringv1.Prometheus
-			C Config
-			R []string
-		}{p, c, ruleConfigMapNames},
+	hash, err := hashstructure.Hash(struct {
+		P monitoringv1.Prometheus
+		C Config
+		R []string `hash:"set"`
+	}{p, c, ruleConfigMapNames},
```

Comment:
sorry for a newbie question: does `ruleConfigMapNames` always maintain the set invariant? Do we make sure this is the case already?

Reply:
Not sure I understand the question correctly: Treating `ruleConfigMapNames` as a set instead of an array is enforced via the struct tag `R []string `hash:"set"``.

Reply:
Sorry for being imprecise: do we make sure that `ruleConfigMapnames` never contains duplicate enries? If they do, they would hash to the same value when treated as a set of strings.

Reply:
Yes we do here: https://github.com/coreos/prometheus-operator/blob/master/pkg/prometheus/rules.go#L241

In addition it is enforced by Kubernetes, as there can not be multiple ConfigMaps with the same name in the same namespace.